src/ldp/http/SparqlUpdateBodyParser.ts

```diff
@@ -31,7 +31,7 @@ export class SparqlUpdateBodyParser extends BodyParser {
     let algebra: Algebra.Operation;
     try {
       const sparql = await readableToString(toAlgebraStream);
-      algebra = translate(sparql, { quads: true });
+      algebra = translate(sparql, { quads: true, baseIRI: request.url });
```

Comment:
We could also pass the extracted IRI along to the other parsers in the `BasicRequestParser`, just like we pass the generated metadata to the body parser.

Reply:
Actually, since we are passing metadata, can we access `metadata.identifier` or so?

Reply:
No, since at that point the metadata has a blank node as base identifier since it also doesn't know what the identifier is. This could also be solved by passing it the result of the target extractor (but less relevant since it will be updated at a later point when needed). This would also not always be correct since in the case of a POST it does not need that identifier (but again, not a problem since it's updated later on).

Reply:
> it also doesn't know what the identifier is.

Why not though? It should be that of the request? (It's `PATCH`, not `POST`.)

Reply:
> Why not though? It should be that of the request? (It's `PATCH`, not `POST`.)

I meant in general, in this case it would be an option yes, but then someone would have to tell the metadata extractor what the identifier is, which is the same problem as the patch parser has here.

https://github.com/solid/community-server/blob/a5bc8d22a9ce028a8bd8b17ca3658832d4f35ec9/src/ldp/http/BasicRequestParser.ts#L39-L42

If we pass the `target` to the other handlers in the code above we could fix this: either pass it to the metadata handler so the body handler can extract it from the metadata, or just pass it to all handlers.

Reply:
Maybe the metadata handler is the one that should call the `targetExtractor` instead; that would solve it.

Reply:
That's also an option. It might feel a bit more obfuscated that the target of the resulting `Operation` is now found by doing `metadata.identifier` but no real strong opinion there.

Reply:
Also true. There seems to be no straightforward solution.

Perhaps we can just _also_ put it in `metadata.identifier`.

Reply:
I also think that one is the least intrusive.